### DISCUSSION TITLE
Meross MTS 200 overheat warning Problem

### DISCUSSION THREAD

**Discussion by Breitec** [krahabb/meross_lan#277] [Q&A] [answered]
I'm excited about your Home Assistant integration for Meross. Unfortunately, I now already have the problem with my second MTS200 that I have a sensor "overheat warning" problem. I can't find a hint in the Meross app itself. Do you know this problem or can someone help me?
![Bildschirmfoto 2023-04-15 um 11 09 59](https://user-images.githubusercontent.com/12207981/232203259-8894f2d4-2071-4d09-aa7d-215f0ee89b8e.png)

**Comment by krahabb** [comment, marked as answer]
Hello @Breitec,
Thank you for your comment!
The mts200 external sensor feature still needs some care though...
[Here](https://github.com/krahabb/meross_lan/discussions/242#discussioncomment-4925160) a similar (likely identical) issue and some deep feedback from @ikoz which initially helped me in understanding the external sensor feature. In the following replies he deeply tries to reverse-engineer the behavior but I'm actually a bit behind (and so meross_lan) on the implementation.
I'll try to 'stick' this on my TODO and quickly come back...a new awesome (I hope) release is coming out soon

**Reply by ikoz** [reply]
Yeah after my testing what I've found is that the `overheat.warning` message, as it comes from the thermostat has 3 possible values:

`0`: external sensor connected, no overheat alert
`1`: external sensor connected, overheat alert
`2`: external sensor not connected

I'm assuming that the integration currently interprets `2` as `On` since its a binary sensor.

A quick fix could be to interpret only `1` as sensor `On` state any other states as `Off`
